### ISSUE TITLE
[BUG] False Alarm on Model Training Memory 

### ISSUE THREAD

**Issue by ShaohongBai** [awslabs/autogluon#1614]
- [X] I have checked that this bug exists on the latest stable version of AutoGluon
- [ ] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**
Model training with GBM and XGB on m5.24xlarge are alerting with insufficient memory, also did not trigger parallel fold training (not sure whether it is related)

**Expected behavior**
Expect no such memory alerts and also be able to have fold trained by parallel

**Screenshots**
![2022-03-16 at 22 15](https://user-images.githubusercontent.com/37364351/158700747-a76b19a0-15c9-40b9-ba44-f524c1530b92.jpg)
![2022-03-16 at 22 25](https://user-images.githubusercontent.com/37364351/158701517-a5cf8994-ef12-49ce-b01b-130112c83c3c.jpg)

**Installed Versions**
------------------
date                 : 2022-03-16
time                 : 22:17:57.442844
python               : 3.8.9.final.0
OS                   : Linux
OS-release           : 5.4.0-1063-aws
Version              : #66~18.04.1-Ubuntu SMP Thu Jan 13 19:45:37 UTC 2022
machine              : x86_64
processor            : x86_64
num_cores            : 96
cpu_ram_mb           : 382673
cuda version         : None
num_gpus             : 0
gpu_ram_mb           : []
avail_disk_size_mb   : 43677

autogluon.common     : 0.4.0
autogluon.core       : 0.4.0
autogluon.features   : 0.4.0
autogluon.tabular    : 0.4.0
autogluon.text       : 0.4.0
autogluon.vision     : 0.4.0
autogluon_contrib_nlp: None
boto3                : 1.20.24
catboost             : 1.0.4
dask                 : 2021.11.2
distributed          : 2021.11.2
fairscale            : 0.4.6
fastai               : 2.5.3
gluoncv              : 0.11.0
lightgbm             : 3.3.2
matplotlib           : 3.5.1
networkx             : 2.7.1
nptyping             : 1.4.4
numpy                : 1.22.3
omegaconf            : 2.1.1
pandas               : 1.3.5
PIL                  : 9.0.1
psutil               : 5.8.0
pytorch_lightning    : 1.5.10
ray                  : 1.8.0
requests             : 2.27.1
scipy                : 1.7.3
sentencepiece        : None
skimage              : 0.19.2
sklearn              : 1.0.2
smart_open           : 5.2.1
timm                 : 0.5.4
torch                : 1.10.2+cu102
torchmetrics         : 0.7.2
tqdm                 : 4.63.0
transformers         : 4.16.2
xgboost              : 1.4.2

**Comment by Innixma** [comment]
Thanks for creating this issue!

This is due to code logic in XGBoost that will log a warning if the model is expected to use more than 20% of available memory. This is probably too low (error occurs at 100%), so we can increase the value to 75% which will silence the warning in your case. This has been updated in #1626.

Regarding parallel folding, we naively assume that parallel folding will be a linear increase in memory usage, and we train 8 folds in parallel. Because the amount of memory expected to be required multiplied by 8 is greater than available system memory, AutoGluon instead trains sequentially. Whether the memory estimates are accurate or not is another matter, currently they are based on heuristic rules.

The current memory estimation rule for XGBoost:

```
    def _estimate_memory_usage(self, X, **kwargs):
        num_classes = self.num_classes if self.num_classes else 1  # self.num_classes could be None after initialization if it's a regression problem
        data_mem_usage = get_approximate_df_mem_usage(X).sum()
        approx_mem_size_req = data_mem_usage * 7 + data_mem_usage / 4 * num_classes  # TODO: Extremely crude approximation, can be vastly improved
        return approx_mem_size_req
```

**Comment by FailedPipeline** [comment]
I am experiencing this same bug in 0.8.2 and 0.6.2 training LightGBM models.
Tons of memory headroom yet training slips into sequential.
Should I make a new ticket?
